### DISCUSSION TITLE
Semantic Emoji Finder thanks to BERT support

### DISCUSSION THREAD

**Discussion by astrowonk** [ggml-org/llama.cpp#5767] [Show and tell]
I was able to use llama.cpp (via the python bindings) to generate BERT embeddings with relatively little memory (compared to `pytorch` and `sentence_transformers`) and allow for [multi-word searches in my semantic emoji finder](https://github.com/astrowonk/emoji_finder/pull/7/). I think a `sentence-transformers` version of this web app would use nearly 1GB of RAM; now implemented it's running in less than <200MB.

I've deployed the changes on the [deployed Dash app](https://marcoshuerta.com/dash/emoji_finder/).

**Comment by ggerganov** [comment]
Very cool!
